third_party/move/move-vm/runtime/src/data_cache.rs

```diff
@@ -58,14 +97,16 @@ impl DataCacheEntry {
 /// and pass it to the Move VM.
 pub struct TransactionDataCache {
     account_map: BTreeMap<AccountAddress, BTreeMap<Type, DataCacheEntry>>,
+    enable_resource_existence_optimization: bool,
```

Comment:
check_resource_existence_without_load or can still have enabled, could also find a better name but I think w.o. loading / no load smt like this is more descriptive than the fact that it's an optimization. In fact, we could add a small comment explaining the optimization aspect. More importantly, we can explain the gating strategy there (seems bottlenecked in create_and_insert_or_upgrade_and_charge_data_cache_entry?)